src/test/javascript/spec/component/exam/manage/exam-management.service.spec.ts

```diff
@@ -82,70 +84,174 @@ describe('Exam Management Service Tests', () => {
+        service.getExamScores(course.id!, mockExam.id!).subscribe((res) => expect(res.body).to.deep.eq(expectedExamScore));
 
         // THEN
         const req = httpMock.expectOne({ method: 'GET', url: `${service.resourceUrl}/${course.id!}/exams/${mockExam.id}/scores` });
+        req.flush(mockExamScore);
+        tick();
+    }));
 
-        // CLEANUP
-        req.flush(mockExam);
-    });
+    it('should get stats for exam assessment dashboard', fakeAsync(() => {
+        // GIVEN
+        const mockExam: Exam = { id: 1 };
+        const mockStatsForDashboard = new StatsForDashboard();
+        const expectedStatsForDashboard = { ...mockStatsForDashboard };
 
-    it('should find all exams for course', () => {
         // WHEN
-        service.findAllExamsForCourse(course.id!).subscribe((res) => expect(res.body).to.equal([]));
+        service.getStatsForExamAssessmentDashboard(course.id!, mockExam.id!).subscribe((res) => expect(res.body).to.deep.eq(expectedStatsForDashboard));
 
         // THEN
-        httpMock.expectOne({ method: 'GET', url: `${service.resourceUrl}/${course.id!}/exams` });
-    });
+        const req = httpMock.expectOne({ method: 'GET', url: `${service.resourceUrl}/${course.id}/exams/${mockExam.id}/stats-for-exam-assessment-dashboard` });
+        req.flush(mockStatsForDashboard);
+        tick();
+    }));
+
+    it('should find all exams for course', fakeAsync(() => {
+        // GIVEN
+        const mockExam: Exam = {
+            id: 1,
+            startDate: undefined,
+            endDate: undefined,
+            visibleDate: undefined,
+            publishResultsDate: undefined,
+            examStudentReviewStart: undefined,
+            examStudentReviewEnd: undefined,
+        };
+        const mockExamResponse = [{ ...mockExam }];
 
-    it('should find all current and upcoming exams', () => {
         // WHEN
-        service.findAllCurrentAndUpcomingExams().subscribe((res) => expect(res.body).to.equal([]));
+        service.findAllExamsForCourse(course.id!).subscribe((res) => expect(res.body).to.deep.equal([mockExam]));
 
         // THEN
-        httpMock.expectOne({ method: 'GET', url: `${service.resourceUrl}/upcoming-exams` });
-    });
+        const req = httpMock.expectOne({ method: 'GET', url: `${service.resourceUrl}/${course.id!}/exams` });
+        req.flush(mockExamResponse);
+        tick();
+    }));
 
-    it('should getExamWithInterestingExercisesForAssessmentDashboard with isTestRun=false', () => {
+    it('find all exams for which the instructors have access', fakeAsync(() => {
         // GIVEN
-        const mockExam: Exam = { id: 1 };
+        const mockExam: Exam = {
+            id: 1,
+            startDate: undefined,
+            endDate: undefined,
+            visibleDate: undefined,
+            publishResultsDate: undefined,
+            examStudentReviewStart: undefined,
+            examStudentReviewEnd: undefined,
+        };
```

Comment:
Can you move this into a function that gives you a mock exam or define it in before each so it does not get repeated? This will help you reduce the codacy errors as well.